koji/flatten-el8/koji-ansible/roles/grobisplitter/files/splitter.py

```diff
@@ -0,0 +1,397 @@
+#!/bin/python3
+
+# Import libraries needed for application to work
+
+import argparse
+import shutil
+import gi
+import gzip
+import librepo
+import hawkey
+import tempfile
+import os
+import subprocess
+import sys
+
+# Look for a specific version of modulemd. The 1.x series does not
+# have the tools we need.
+try:
+    gi.require_version('Modulemd', '2.0')
+    from gi.repository import Modulemd
+except:
+    print("We require newer vesions of modulemd than installed..")
+    sys.exit(0)
+    
+mmd = Modulemd
+
+# This code is from Stephen Gallagher to make my other caveman code
+# less icky.
+def _get_latest_streams (mymod, stream):
+    """
+    Routine takes modulemd object and a stream name.
+    Finds the lates stream from that and returns that as a stream
+    object. 
+    """
+    all_streams = mymod.search_streams(stream, 0)
+    latest_streams = mymod.search_streams(stream,
+                                          all_streams[0].props.version) 
+    
+    return latest_streams
+    
+def _get_repoinfo(directory):
+    """
+    A function which goes into the given directory and sets up the
+    needed data for the repository using librepo.
+    Returns the LRR_YUM_REPO
+    """
+    with tempfile.TemporaryDirectory(prefix='elsplit_librepo_') as lrodir:
+        h = librepo.Handle()
+        h.setopt(librepo.LRO_URLS, ["file://%s" % directory])
+        h.setopt(librepo.LRO_REPOTYPE, librepo.LR_YUMREPO)
+        h.setopt(librepo.LRO_DESTDIR, lrodir)
+        h.setopt(librepo.LRO_LOCAL, True)
+        h.setopt(librepo.LRO_IGNOREMISSING, False)
+        r = h.perform()
+        return r.getinfo(librepo.LRR_YUM_REPO)
+
+def _get_hawkey_sack(repo_info):
+    """
+    A function to pull in the repository sack from hawkey.
+    Returns the sack.
+    """
+    hk_repo = hawkey.Repo("")
+    hk_repo.filelists_fn = repo_info["filelists"]
+    hk_repo.primary_fn = repo_info["primary"]
+    hk_repo.repomd_fn = repo_info["repomd"]
+
+    primary_sack = hawkey.Sack()
+    primary_sack.load_repo(hk_repo, build_cache=False)
+    
+    return primary_sack
+
+def _get_filelist(package_sack):
+    """
+    Determine the file locations of all packages in the sack. Use the
+    package-name-epoch-version-release-arch as the key.
+    Returns a dictionary.
+    """
+    pkg_list = {}
+    for pkg in hawkey.Query(package_sack):
+        nevr="%s-%s:%s-%s.%s"% (pkg.name,pkg.epoch,pkg.version,pkg.release,pkg.arch)
+        pkg_list[nevr] = pkg.location
+    return pkg_list
+
+def _parse_repository_non_modular(package_sack, repo_info, modpkgset):
+    """
+    Simple routine to go through a repo, and figure out which packages
+    are not in any module. Add the file locations for those packages
+    so we can link to them.
+    Returns a set of file locations.
+    """
+    sack = package_sack
+    pkgs = set()
+
+    for pkg in hawkey.Query(sack):
+        if pkg.location in modpkgset:
+            continue
+        pkgs.add(pkg.location)
+    return pkgs
```

Comment:
```suggestion
    return {pkg.location for pkg in hawkey.Query(sack) if pkg.location in modpkgset}
```

Or if you modify `_get_modular_pkgset` to return a set:
```suggestion
    return {pkg.location for pkg in hawkey.Query(sack)} & modpkgset
```

Reply:
I'd prefer not to deviate from their script more than necessary at this time.